plugins/beacon/ios/Tests/BaseBeaconPluginTests.swift

```diff
@@ -136,4 +136,105 @@ class BaseBeaconPluginTests: XCTestCase {
         ))
         wait(for: [expectation], timeout: 2)
     }
+    
+    func testCancelSpecificBeaconsUsingHooks() {
+        let context = JSContext()!
+        JSUtilities.polyfill(context)
+        
+        let cancelBeaconHandler = expectation(description: "Cancel Beacon Handler called")
+        let handlerExpectation = expectation(description: "Handler called")
+        handlerExpectation.isInverted = true
+        
+        let handler = MockHandler()
+        
+        let plugin = BaseBeaconPlugin<DefaultBeacon>(plugins: []) { (beacon) in
+            if handler.calls.isEmpty {
+                handler.handle(beacon.viewId!, beacon.data)
+                handlerExpectation.fulfill()
+            }
+        }
+        
+        plugin.context = context
+        plugin.setup(context: context)
+        
+        plugin.beacon(assetBeacon: AssetBeacon(
+            action: BeaconAction.clicked.rawValue,
+            element: BeaconElement.button.rawValue,
+            asset: BeaconableAsset(id: "test"),
+            data: .string(data: "example")
+        ))
+        
+        guard let hooks = plugin.hooks else {
+            XCTFail("Hooks are not initialized")
+            return
+        }
+        
+        hooks.cancelBeacon.tap { (arg1: JSValue, arg2: JSValue) -> Bool in
+            cancelBeaconHandler.fulfill()
```

Comment:
I'm surprised this is working, the hook should get called by the underlying JS plugin when `beacon` is called, but you are tapping after calling `beacon`?

Reply:
fixed